### ISSUE TITLE
Moving cards between columns in mobile view is not obvious and can lead to getting a UI bug.

### ISSUE THREAD

**Issue by danielwilson654** [wekan/wekan#3566]
## Issue I still haven't figured out how to move cards between columns in mobile mode. I'm no that thick.. I think. Anyway, if I can't figure it out after 2 minutes I expect the users I want to roll this out to wont be able to figure out either.

**Server Setup Information**: **Docker**
* Did you test in newest Wekan?: **Yes**
* Wekan only works on newest desktop Firefox/Chromium/Chrome/Edge/Chromium Edge and mobile Chrome. What webbrowser version are you using? : **Firefox**

**Problem description**:
- *REQUIRED: Add recorded animated gif about how it works currently, and screenshot mockups how it should work. Use peek to record animgif in Linux https://github.com/phw/peek* 
![recording(1)](https://user-images.githubusercontent.com/17303371/107718665-8b6a8500-6d2a-11eb-9de7-cc561441d91e.gif)

- *Explain steps how to reproduce* look at card wall on mobile. Try drag card to the right of the screen.

**Comment by mfilser** [comment]
1. Open the card you want to move
2. at the top click the Hamburger icon
3. select "Move Card"
4. fill out the popup
5. click "Done"

**Comment by danielwilson654** [comment]
Thanks for the feedback, however, I don't think this should be closed.

The method mentioned above is not obvious. When moving from a desktop, you get used to moving the cards vertically as well as horizontally. The experience should be as intuitive when on mobile. I know now how to do it, but as a UI/UX designer I know it's going to be a stumbling block for others. 

I might be able to mock up a better solution later, but for now could you leave this open?

**Comment by mfilser** [comment]
That's right, it could be better, but also at small screens there are less options.
Anyway, i will think there could be a better doing to move cards to another list. 
I am thinking of an easier method...

**Comment by xet7** [comment]
@danielwilson654 

Wekan mobile phone interface is optimized for small screen touch screen usage. Ideas welcome how to make mobile phone interface better.

If there is more screen space, like at tablet screen, Wekan shows desktop style user interface.

**Comment by xet7** [comment]
Well, if someone is used to drag-drop desktop interface, and someone has time to contribute to Wekan to enable desktop interface on mobile screen, let's tell that secret story about how that mobile web interface works.

There is code isMiniScreen here:
https://github.com/wekan/wekan/blob/master/client/lib/utils.js#L204

But, changing that to `return false` is not enough, because with this search:
```
cd wekan
./find.sh 800px
```
You find this kind of code:
```
@media screen and (max-width: 800px)
```
That stylus language `.styl` is that special kind of css+extras, that you can have also Javascript/Jade style code in those `.styl` files. So there can be this kind of code:
```
if isMiniScreen
```
For other widths, you can search with:
```
./find.sh max-width
```
So each of those widths should be replaced with some function call, similar to `isMiniScreen`.

There is also this kind of checking code:
```
if (Utils.isMiniScreen() || showDesktopDragHandles) {
```
So it checks when there is DesktopDragHandles.

There is also `.jade` templates where is code like this:
```
{{#unless isMiniScreen}}{{_ 'people-number'}}{{/unless}} #{peopleNumber}
```
And at Javascript code like this:
```
./client/components/main/editor.js:    const isSmall = Utils.isMiniScreen();
```
But usually, if in `wekan/client/` directory if somewhere isMiniScreen does not work, it is added to be working like this in Javascript `.js` file:
```
function isMiniScreen() {
  return Utils.isMiniScreen();
}
```
And then it can be used in `.jade` and `.styl` files like this:
```
if isMiniScreen
```
If needed, it would be also possible to check for device orientation, like mentioned at [Stackoverflow](https://stackoverflow.com/questions/4917664/detect-viewport-orientation-if-orientation-is-portrait-display-alert-message-ad):
```
if (window.matchMedia("(orientation: portrait)").matches) {
   // you're in PORTRAIT mode
}

if (window.matchMedia("(orientation: landscape)").matches) {
   // you're in LANDSCAPE mode
}
```
If there were button for changing between mobile and desktop interface, that could be added to:
- board header at `wekan/client/components/boards/`
- to database, new profile model field at `wekan/models/users.js`

## About mobile web and touch screen interface in general:
1. Drag handles are very important. That way, drag handle can drag, and other area can be used to pinch-zoom and change what part of board to view. So it would be preferred to have drag handles enabled at touch screen, because otherwise you can accidentally move any card anywhere because all card areas would be for dragging immediately.
2. Previously when I tried, at Trello:
  -  web version drag-drop does not work well on mobile webbrowser
  - mobile app can delay-drag cards. But because AFAIK delay-drag does not work in webbrowser, for Wekan drag handles make possible immediate dragging, without delay.
3. If desktop mode would be enabled in mobile phone screen, it would work, although that way it would be very often necessary to use pinch-zoom because mobile screen is so small.
4. Current mobile phone interface is very usable, so I hope changing to desktop inferface would be optional, and mobile phone interface would not be completely disabled.

**Comment by 1ubuntuuser** [comment]
@xet7 Thank you! What you have done is given others a leg up to get something done! This is obviously better then just adding it to the to-do list of things you have to do... Those code examples are very useful.

**Comment by xet7** [comment]
@1ubuntuuser 

Thanks! This gave me an idea, so I added links to some stories to [Developer Docs](https://github.com/wekan/wekan/wiki/Developer-Documentation#stories-about-how-wekan-works-and-developing-wekan).

I have written [most of the docs at wiki](https://github.com/wekan/wekan/wiki).

**Comment by 1ubuntuuser** [comment]
@xet7 That is really valuable! I had a look through the links. If you come across any other nuggets in future please add them there. 
Is there a write up hanging around somewhere with details on how to setup a dev env for wekan with Visual Studio Code?

**Comment by xet7** [comment]
@1ubuntuuser 

I reordered Developer Docs page so hopefully it's now easier to find here https://github.com/wekan/wekan/wiki/Developer-Documentation#vscode--vscodium
